git-gui.sh

```diff
@@ -624,7 +624,10 @@ proc git_write {args} {
 }
 
 proc githook_read {hook_name args} {
-	set pchook [gitdir hooks $hook_name]
+	if {[catch {set hooksdir [git config core.hooksPath]}]} {
```

Comment:
There is a `get_config` function that uses a cached config (you need to use lower-case keys for that, though, IIRC).

In any case, this is duplicating the logic in Git's `config.c`, therefore a bit fragile, besides, if Git GUI is used with Git <v2.9.0 it would be inconsistent with Git's idea of the hooks path.

How about something like this instead (completely untested, you likely have to fix a few bits here and there):

```diff
diff --git a/git-gui/git-gui.sh b/git-gui/git-gui.sh
index 37c1c5d227b..3067a3b000a 100755
--- a/git-gui/git-gui.sh
+++ b/git-gui/git-gui.sh
@@ -624,7 +624,11 @@ proc git_write {args} {
 }
 
 proc githook_read {hook_name args} {
-	set pchook [gitdir hooks $hook_name]
+	if {[package vcompare $::_git_version 2.5.0] >= 0} {
+		set pchook [git rev-parse --git-path "hooks/$hook_name"]
+	} else {
+		set pchook [gitdir hooks $hook_name]
+	}
 	lappend args 2>@1
 
 	# On Windows [file executable] might lie so we need to ask
```

I offered the same patch [on the Git mailing list](https://public-inbox.org/git/alpine.DEB.2.21.1.1706141516540.171564@virtualbox/), together with a suggestion for an improved commit message.

Reply:
How about making it so by amending your patch and force-pushing, just in case that Pat returns from his absence and wants to merge the open PRs?